### PR TITLE
log: fix elastic version detection

### PR DESCRIPTION
**Pull request by huanggze**
Signed-off-by: huanggze <loganhuang@yunify.com>

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change

/kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

```
If elastic version number is not provided, it always create esclient for v6. 
```

**Special notes for reviewers**:
```
```

**Additional documentation, usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/simple/client/elasticsearch/esclient.go [hunk 1]
```diff
@@ -60,12 +60,11 @@ type ElasticSearchClient struct {
 }
 
 func NewLoggingClient(options *ElasticSearchOptions) (*ElasticSearchClient, error) {
-	version := "6"
 	esClient := &ElasticSearchClient{}
 
 	if options.Version == "" {
 		var err error
-		version, err = detectVersionMajor(options.Host)
+		options.Version, err = detectVersionMajor(options.Host)
```
Comment (zryfish) [inline pkg/simple/client/elasticsearch/esclient.go]: not a good practice, since you've changed options value

Comment (huanggze) [reply pkg/simple/client/elasticsearch/esclient.go]: the same here, `options.Index = options.LogstashPrefix`. Should we refactor all of them?

Comment (huanggze) [reply pkg/simple/client/elasticsearch/esclient.go]: https://github.com/kubesphere/kubesphere/pull/799/files#diff-28320d9ea55aa094d44988b9161bbe6aR73-R77

Comment (zryfish) [reply pkg/simple/client/elasticsearch/esclient.go]: yes, please

**Pull request merged**